src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -946,6 +959,18 @@ private JsonRpcMessage HandleInvocationTaskResult(JToken id, Task t)
 
             if (t.IsFaulted)
             {
+                if (t.Exception != null)
+                {
+                    if (t.Exception.InnerException != null && this.IsFatalException(t.Exception.InnerException))
+                    {
+                        ExceptionDispatchInfo.Capture(t.Exception.InnerException).Throw();
+                    }
+                    else if (this.IsFatalException(t.Exception))
```

Comment:
This seems odd that you're calling `IsFatalException` *twice* for ultimately the same exception. Can we get it to just be called once? (and then perhaps add assertions to your tests to ensure it's only invoked once so it doesn't regress).